errors/react-hydration-error.md

```diff
@@ -0,0 +1,17 @@
+# React Hydration Error
+
+#### Why This Error Occurred
```

Comment:
```suggestion
## Why This Error Occurred
```

Reply:
This is following the template we use here https://raw.githubusercontent.com/vercel/next.js/canary/errors/template.md and existing errors follow these headings [here](https://raw.githubusercontent.com/vercel/next.js/canary/errors/experimental-jest-transformer.md) if we want to change this we should update the template and the existing error documents.

Reply:
This is just very odd and not even close to being semantically correct 😅. Lets merge this and create an issue to change the template 😊